stravalib/client.py

```diff
@@ -1760,53 +1902,79 @@ def next(self):
 
 class ActivityUploader(object):
     """
-    The "future" object that holds information about an activity file upload and can
-    wait for upload to finish, etc.
+    The "future" object that holds information about an activity file
+    upload and can wait for upload to finish, etc.
+
     """
 
     def __init__(self, client, response, raise_exc=True):
         """
-        :param client: The :class:`stravalib.client.Client` object that is handling the upload.
-        :type client: :class:`stravalib.client.Client`
-
-        :param response: The initial upload response.
-        :type response: Dict[str,Any]
-
-        :param raise_exc: Whether to raise an exception if the response
-                  indicates an error state. (default True)
-        :type raise_exc: bool
+        client: `stravalib.client.Client`
+            The :class:`stravalib.client.Client` object that is handling the
+            upload.
+        response: Dict[str,Any]
+            The initial upload response.
+        raise_exc: bool
+            Whether to raise an exception if the response
+            indicates an error state. (default True)
         """
         self.client = client
         self.response = response
         self.update_from_response(response, raise_exc=raise_exc)
 
     @property
     def photo_metadata(self):
-        """
-        photo metadata for the activity upload response, if any.
+        """photo metadata for the activity upload response, if any.
         it contains a pre-signed uri for uploading the photo.
-        Note1: This is only available after the upload has completed.
-        Note2: available only for partner apps (not for regular account)
+
+        Notes
+        -----
+        * This is only available after the upload has completed.
+        * This metadata is only available for partner apps. If you have a
+        regular / non partner related Strava app / account it will not work.
+
         """
 
         warn_attribute_unofficial("photo_metadata")
 
         return self._photo_metadata
 
+    # TODO: what is value here? perhaps an int representing a photo id?
```

Comment:
@jsamoocha i had a question about this as well. i suspect photo_metadata would be an `int` representing the photo id value. but i haven't worked with photos at all yet to know for sure so i wanted to ask before updating.

Reply:
Here's the code where it's set (which was outside of `__init__()`, so not straightforward to find its type):

```python
# undocumented field, it contains pre-signed uri to upload photo to
self._photo_metadata: Optional[List[Dict]] = response.get(
    "photo_metadata"
)
```

Reply:
ok i looked at that too quickly. I see the Optional is a typing object now. so i have list of dictionaries or `None` as the type here.